ui/src/app/pages/login/account-creation/account-creation.tsx

```diff
@@ -0,0 +1,665 @@
+      return true;
+    }
+    // Include alphanumeric characters, -'s, _'s, apostrophes, and single .'s in a row.
+    if (username.includes('..') || username.endsWith('.')) {
+      return true;
+    }
+    return !(new RegExp(/^[\w'-][\w.'-]*$/).test(username));
+  }
+
+  usernameChanged(value: string): void {
+    this.updateProfile('username', value);
+    const {username} = this.state.profile;
+    if (username === '') {
+      return;
+    }
+    this.setState({usernameConflictError: false});
+    // TODO: This should use a debounce, rather than manual setTimeout()s.
+    clearTimeout(this.usernameCheckTimeout);
+    this.setState({usernameCheckInProgress: true});
+    this.usernameCheckTimeout = setTimeout(() => {
+      if (!username.trim()) {
+        this.setState({usernameCheckInProgress: false});
+        return;
+      }
+      profileApi().isUsernameTaken(username)
+        .then((body) => {
+          this.setState({usernameCheckInProgress: false, usernameConflictError: body.isTaken});
+        })
+        .catch((error) => {
+          console.log(error);
+          this.setState({usernameCheckInProgress: false});
+        });
+    }, 300);
+  }
+
+  updateProfile(attribute: string, value: string) {
+    if (attribute === 'contactEmail') {
+      this.setState({invalidEmail: false});
+    }
+    const newProfile = this.state.profile;
+    newProfile[attribute] = value;
+    this.setState(({profile}) => ({profile: fp.set(attribute, value, profile)}));
+  }
+
+  // The strings will be changed to ENUM value once survey page is done
+  updateAffiliationRoles(affiliation) {
+    this.setState({showAffiliationRole: false, showAffiliationOther: false,
+      affiliation: affiliation});
+    if (affiliation === 'industry') {
+      this.setState({rolesOptions: Options.industryRole, showAffiliationRole: true});
+    } else if (affiliation === 'educationalInstitution') {
+      this.setState({rolesOptions: Options.eductionRole, showAffiliationRole: true});
+    } else if (affiliation === 'freeText') {
+      this.setState({showAffiliationOther: true});
+      return;
+    }
+    this.selectAffiliationRoles(this.state.affiliationRole);
+  }
+
+  selectAffiliationRoles(role) {
+    this.setState({affiliationRole: role});
+    if (role === 'freeText') {
+      this.setState({showAffiliationOther: true});
+    } else {
+      this.setState({showAffiliationOther: false});
+    }
+  }
+
+  // This will be deleted once enableAccountPages is set to true for prod
+  validate() {
+    const {profile} = this.state;
+    const requiredFields =
+      [profile.givenName, profile.familyName, profile.username, profile.contactEmail,
+        profile.currentPosition, profile.organization, profile.areaOfResearch];
+    if (requiredFields.some(isBlank)) {
+      this.setState({showAllFieldsRequiredError: true});
+      return;
+    }
+  }
+
+  validateAccountCreation() {
+    const {profile: {
```

Comment:
Nit: For spacing, do you mind putting `profile` on a newline? This is sorta weirdly spaced with everything else in the object.